p2p/src/sync/peer.rs

```diff
@@ -47,33 +47,33 @@ where
     T: NetworkingService,
 {
     /// Unique peer ID
-    peer_id: T::PeerId,
+    _peer_id: T::PeerId,
 
     /// State of the peer
     state: PeerSyncState,
 
-    /// Locator that was sent to the peer
-    /// Used to verify header response and pick unknown headers
-    locator: Vec<BlockHeader>,
-
     /// List of block headers indicating which blocks
     /// still need to be downloaded from the remote peer
     work: VecDeque<BlockHeader>,
 }
 
-impl<T> PeerContext<T>
-where
-    T: NetworkingService,
-{
-    pub fn new(peer_id: T::PeerId, locator: Vec<BlockHeader>) -> Self {
+impl<T: NetworkingService> PeerContext<T> {
+    pub fn new(_peer_id: T::PeerId) -> Self {
         Self {
-            peer_id,
-            locator,
+            _peer_id,
             state: PeerSyncState::Unknown,
             work: VecDeque::new(),
         }
     }
 
+    pub fn new_with_locator(_peer_id: T::PeerId, locator: Vec<BlockHeader>) -> Self {
```

Comment:
I think there should be a separate type for locators... maybe later.

Reply:
I'll add an entry to my TODO list